---
title: C6318 | Документация Майкрософт
ms.custom: ''
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C6318
helpviewer_keywords:
- C6318
ms.assetid: 3284a83e-bb8e-461c-adcc-cfc66ceea05e
caps.latest.revision: 15
author: mikeblome
ms.author: mblome
manager: ghogen
ms.openlocfilehash: 58eb6f9f41eac282e2b7cce4477d204e88419152
ms.sourcegitcommit: af428c7ccd007e668ec0dd8697c88fc5d8bca1e2
ms.translationtype: MT
ms.contentlocale: ru-RU
ms.lasthandoff: 11/16/2018
ms.locfileid: "51770768"
---
# <a name="c6318"></a>C6318
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

Предупреждение C6318: неправильно определенный блок __try /\__except: использование константы EXCEPTION_CONTINUE_SEARCH или другой константы с нулевым результатом в выражении фильтра исключений. Код в блоке обработки исключений не выполняется.  
  
 Это предупреждение указывает, что при возникновении исключения в защищенном блоке обработчика структурированных исключений, исключение не будет обработано, так как константа `EXCECPTION_CONTINUE_SEARCH` используется в выражении фильтра исключений.  
  
 Этот код эквивалентен защищенный блок без блока обработки исключений, так как в блоке обработчика не выполняется.  
  
## <a name="example"></a>Пример  
 В следующем коде создается данное предупреждение:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except (EXCEPTION_CONTINUE_SEARCH) // warning 6318  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 Чтобы устранить это предупреждение, используйте следующий код:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except( (GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION) ?  
                EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )  
  {  
    puts("Access violation");  
  }  
}   
```  
  
## <a name="see-also"></a>См. также  
 [Оператор try-except](http://msdn.microsoft.com/library/30d60071-ea49-4bfb-a8e6-7a420de66381)



